src/pages/index.js

```diff
@@ -20,100 +24,178 @@ import {
-/*                     Functions                     */
+/*                 Event Handlers                    */
 /*---------------------------------------------------*/
 
-function createCard(cardData) {
-  const name = cardData.name;
-  const altName = cardData.name;
-  const link = cardData.link;
-  const card = new Card(
-    { name, altName, link },
-    "#card-template",
-    handleImageClick
-  );
-  return card.getView();
+function handleAvatarFormSubmit(formValues) {
+  const profilePicObject = api.patchProfileAvatar(formValues.url);
+  changeSubmitButton(avatarModalSubmitButton);
+  profilePicObject.then((object) => {
+    avatarPopup.handleAvatarChange(object.avatar);
+  });
+  avatarPopup.close();
+  profileAvatarFormValidator.disableButton();
 }
 
-/*---------------------------------------------------*/
-/*                 Event Handlers                    */
-/*---------------------------------------------------*/
-
 function handleProfileFormSubmit(formValues) {
+  api.patchProfile(formValues.name, formValues.bio);
+  changeSubmitButton(editModalSubmitButton);
   userInfo.setUserInfo(formValues.name, formValues.bio);
   profilePopup.close();
   editProfileFormValidator.disableButton();
 }
 
 function handleAddCardFormSubmit(formValues) {
   const name = formValues.title;
-  const altName = formValues.title;
   const link = formValues.url;
-  const cardElement = createCard({ name, altName, link });
-  section.addItem(cardElement);
+  const cardApiObject = api.postCards({ name, link });
+  changeSubmitButton(addModalSubmitButton);
+  cardApiObject.then((object) => {
+    const element = createCard(object);
+    section.addItem(element);
+  });
   cardPopup.close();
 }
 
 function handleImageClick(data) {
   popupImg.open(data);
+  popupImg.setEventListeners();
+}
+
+function handleConfirmModal(data) {
+  trashModalSubmitButton.textContent = "Yes";
+  trashConfirmPopup.open();
+  trashConfirmPopup.setSubmitFunction(() => {
+    handleDeleteConfirmModal(data.apiData.name);
+    data.removeCard(data.cardElement);
+  });
+}
+
+function handleDeleteConfirmModal(title) {
+  uniqueCards.then((cards) => {
+    cards.forEach((card) => {
+      if (card.name === title) {
+        api.deleteCard(card._id);
+        changeSubmitButton(trashModalSubmitButton);
+      }
+    });
+  });
+
+  trashConfirmPopup.close();
+}
+
+function handleLikeIconClick(data) {
+  if (data.apiData.isLiked) {
+    api.deleteCardLike(data.apiData._id);
+    data.changeHeartIcon(data);
+  } else {
+    api.putCardLike(data.apiData._id);
+    data.changeHeartIcon(data);
+  }
 }
 
 /*---------------------------------------------------*/
 /*                 Event Listeners                   */
 /*---------------------------------------------------*/
 
+profilePenIcon.addEventListener("click", () => {
+  avatarModalSubmitButton.textContent = "Save";
+  avatarPopup.open();
+  profileAvatarFormValidator.disableButton();
+  avatarPopup.setEventListeners();
```

Comment:
* `.setEventListeners()` should be removed from `opening` because it's a **public** method that should be called only **once** in `index.js` otherwise you add listeners again and again to the same elements. This may cause a **`memory leak`**